node/src/archive/sqlite.rs

```diff
@@ -82,6 +82,19 @@ impl Archive {
         Ok(events.json_contract_events)
     }
 
+    /// Fetch the last finalized block height and block hash
+    pub async fn fetch_last_finalized_block(&self) -> Result<(u64, String)> {
```

Comment:
Ditto

Reply:
same here a failure is an internal server error.